align/cell_fabric/gen_lef.py

```diff
@@ -2,138 +2,76 @@
 import json
 import collections
 
-def gen_lef_data(data, fp, macro_name, cell_pin, bodyswitch):
-    def s(x):
-        return "%.4f" % (x/10000.0)
+def lef_from_layout_d(layout_d, fp, out_lef, cell_pin, bodyswitch, blockM, *, exclude_layers, Scale_factor, m1pitch, m2pitch):
 
-    fp.write("MACRO %s\n" % macro_name)
-    fp.write("  ORIGIN 0 0 ;\n")
-    fp.write("  FOREIGN %s 0 0 ;\n" % macro_name)
+    bbox = layout_d['bbox']
 
-    fp.write("  SIZE %s BY %s ;\n" % (s(data['bbox'][2]), s(data['bbox'][3])))
+    assert (bbox[3]-bbox[1]) % m2pitch == 0, f"Cell height not a multiple of the grid {bbox}"
```

Comment:
This check is okay for now but will not work for nonuniform grids. The flow should not assume a uniform grid for metals or only assume for the lowest metal layer (preferably poly).